ICAFIX/ApplyAutoRecleanPipeline.sh

```diff
@@ -0,0 +1,209 @@
+    --python-singularity-mount-path="${PythonSingularityMountPath}" \
+    --python-interpreter="${PythonInterpreter}" \
+    --model-to-use="$ModelToUse" \
+    --vote-threshold="$VoteThresh" \
+    --reclassify-as-signal-file="$ReclassifyAsSignalFile" \
+    --reclassify-as-noise-file="$ReclassifyAsNoiseFile"
+
+# apply the reclassification results
+log_Msg "Begin to apply the reclassification results..."
+IFS='@' read -a fMRINamesArray <<<"$fMRINames"
+
+# check SR or MR FIX
+if [ ! ${MRFixConcatName} = "" ] ; then
+    fMRINameToUse=${MRFixConcatName}
+else
+    fMRINameToUse=""
+    for fMRIName in "${fMRINamesArray[@]}"
+    do
+        fMRINameToUse+=" $fMRIName"
+    done
+    # Remove the leading space
+    fMRINameToUse="${fMRINameToUse:1}"
+fi
+
+for fMRIName in ${fMRINameToUse} ; do
+    "$HCPPIPEDIR"/ICAFIX/ApplyHandReClassifications.sh \
+        --study-folder="$StudyFolder" \
+        --subject="$Subject" \
+        --fmri-name="$fMRIName" \
+        --high-pass="$HighPass" \
+        --reclassify-as-signal-file="$ReclassifyAsSignalFile" \
+        --reclassify-as-noise-file="$ReclassifyAsNoiseFile" \
+        --matlab-run-mode="$MatlabMode"
+done
+
+# reapply fix
+log_Msg "Begin to reapply FIX..."
+
+# default params
+# motion regression or not
+MotionReg=FALSE
+
+# clean up intermediates
+DeleteIntermediates=FALSE
+
+# reapply fix main processing + add rclean substring
+if [ -z ${MRFixConcatName} ]; then
+    # Single Run
+    for fMRIName in "${fMRINamesArray[@]}"; do
+
+        # MSMAll
+        "$HCPPIPEDIR"/ICAFIX/ReApplyFixPipeline.sh \
+            --path="$StudyFolder" \
+            --subject="$Subject" \
+            --fmri-name="$fMRIName" \
+            --high-pass="$HighPass" \
+            --reg-name="MSMAll" \
+            --low-res-mesh="$LowResMesh" \
+            --matlab-run-mode="$MatlabMode" \
+            --motion-regression="${MotionReg}" \
+            --delete-intermediates="${DeleteIntermediates}" \
+            --clean-substring="clean_rclean"
+
+        # Volume+MSMSulc
+        "$HCPPIPEDIR"/ICAFIX/ReApplyFixPipeline.sh \
+            --path="$StudyFolder" \
+            --subject="$Subject" \
+            --fmri-name="$fMRIName" \
+            --high-pass="$HighPass" \
+            --reg-name="NONE" \
+            --low-res-mesh="$LowResMesh" \
+            --matlab-run-mode="$MatlabMode" \
+            --motion-regression="${MotionReg}" \
+            --delete-intermediates="${DeleteIntermediates}" \
+            --clean-substring="clean_rclean"
+
+    done
+else 
+
+    #MR FIX config support for non-HCP settings
+    config=""
+    processingmode="HCPStyleData"
+
+    SubjfMRINames=""
+    for fMRIName in "${fMRINamesArray[@]}"; do
+        if [[ -e "$StudyFolder/$Subject/MNINonLinear/Results/${fMRIName}/${fMRIName}_Atlas_MSMAll.dtseries.nii" ]]; then
+            SubjfMRINames+="@${fMRIName}"
+        fi
+    done
+    # Remove the leading @
+    SubjfMRINames="${SubjfMRINames:1}"
```

Comment:
```suggestion
    SubjfMRINames=$(IFS='@'; echo "${fMRINamesArray[*]}")
```

Reply:
Hi Tim, this suggestion doesn't seem to check if the fMRI runs exist.

Reply:
Ah, sorry, I missed that.  I guess your solution isn't bad if you only need the list as a string, but you could do:
```
fMRINamesExist=()
<loop, test>
    fMRINamesExist+=("$fMRIName")
```
and then the IFS trick.